climada_petals/hazard/rf_glofas/test/test_river_flood_computation.py

```diff
@@ -0,0 +1,381 @@
+import unittest
+from unittest.mock import patch, MagicMock, DEFAULT
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+import xarray as xr
+import xarray.testing as xrt
+import geopandas as gpd
+import pandas.testing as pdt
+
+from climada.util.constants import DEF_CRS
+from climada_petals.hazard.rf_glofas.river_flood_computation import (
+    _maybe_open_dataarray,
+    RiverFloodInundation,
+)
+
+
+class TestMaybeOpenDataArray(unittest.TestCase):
+    """Check the maybe_open_dataarray context manager"""
+
+    def setUp(self):
+        """Create the input array"""
+        self.tempdir = TemporaryDirectory()
+        self.arr_1 = xr.DataArray(data=[0], coords={"dim": [0]})
+        self.arr_2 = xr.DataArray(data=[1], coords={"dim": [1]})
+        self.filename = Path(self.tempdir.name) / "file.nc"
+        self.arr_2.to_netcdf(self.filename)
+
```

Comment:
Note the [docs](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory):

> On completion of the context or destruction of the temporary directory object, the newly created temporary directory and all its contents are removed from the filesystem. [...]
> The directory can be explicitly cleaned up by calling the cleanup() method.
> [...] errors will be raised in whatever context cleanup occurs (the cleanup() call, exiting the context manager, when the object is garbage-collected or during interpreter shutdown).

As far as I understand, the directory is cleaned up when the `TemporaryDirectory` object is destroyed or garbage-collected, which should happen the next time `setUp` is called or the entire test object is deleted. Explicitly calling cleanup should not be required. (But it might be good practice nonetheless?)

Reply:
Whatever you do, get rid of the warnings :D

Reply:
@tovogt What warnings are you referring to? On Python 3.9, running pytest on the module I only get the `RuntimeWarning: divide by zero encountered in divide` you mentioned elsewhere. All other warnings I get are not related to my module, as far as I can see.

Reply:
```shell
$ python --version
Python 3.9.16
$ python -m unittest climada_petals.hazard.rf_glofas.test.test_rf_glofas.TestTransformOps.test_save_file
/home/tovogt/.local/share/miniconda3/envs/climada_env_3.9/lib/python3.9/site-packages/dask/dataframe/_pyarrow_compat.py:17: FutureWarning: Minimal version of pyarrow will soon be increased to 14.0.1. You are using 11.0.0. Please consider upgrading.
  warnings.warn(
/home/tovogt/.local/share/miniconda3/envs/climada_env_3.9/lib/python3.9/tempfile.py:821: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/tmpxkfriv_i'>
  _warnings.warn(warn_message, ResourceWarning)
.
----------------------------------------------------------------------
Ran 1 test in 0.003s

OK
```

Reply:
My interpretation is that the whole point of `TemporaryDirectory` are the automatic cleanup features. If you don't want to make use of it, there is tempfile.mkdtemp which does the same without the automatic cleanup.

Reply:
See https://github.com/CLIMADA-project/climada_petals/pull/64#discussion_r1418964664